scripts/aem.js

```diff
@@ -129,211 +282,150 @@ export async function loadScript(src, attrs) {
-    .map((k) => ICONS_CACHE[k])
-    .filter((v) => !v.styled)
-    .map((v) => v.html)
-    .join('\n');
-  svgSprite.innerHTML += symbols;
-
-  icons.forEach((span) => {
-    const iconName = Array.from(span.classList).find((c) => c.startsWith('icon-')).substring(5);
-    const parent = span.firstElementChild?.tagName === 'A' ? span.firstElementChild : span;
-    // Styled icons need to be inlined as-is, while unstyled ones can leverage the sprite
-    if (ICONS_CACHE[iconName].styled) {
-      parent.innerHTML = ICONS_CACHE[iconName].html;
-    } else {
-      parent.innerHTML = `<svg xmlns="http://www.w3.org/2000/svg"><use href="#icons-sprite-${iconName}"/></svg>`;
-    }
   });
 }
 
 /**
- * Gets placeholders object.
- * @param {string} [prefix] Location of placeholders
- * @returns {object} Window placeholders object
+ * Add <img> for icon, prefixed with codeBasePath and optional prefix.
+ * @param {span} [element] span element with icon classes
+ * @param {string} [prefix] prefix to be added to icon the src
  */
-export async function fetchPlaceholders(prefix = 'default') {
-  window.placeholders = window.placeholders || {};
-  const loaded = window.placeholders[`${prefix}-loaded`];
-  if (!loaded) {
-    window.placeholders[`${prefix}-loaded`] = new Promise((resolve, reject) => {
-      fetch(`${prefix === 'default' ? '' : prefix}/placeholders.json`)
-        .then((resp) => {
-          if (resp.ok) {
-            return resp.json();
-          }
-          throw new Error(`${resp.status}: ${resp.statusText}`);
-        }).then((json) => {
-          const placeholders = {};
-          json.data
-            .filter((placeholder) => placeholder.Key)
-            .forEach((placeholder) => {
-              placeholders[toCamelCase(placeholder.Key)] = placeholder.Text;
-            });
-          window.placeholders[prefix] = placeholders;
-          resolve();
-        }).catch((error) => {
-          // error loading placeholders
-          window.placeholders[prefix] = {};
-          reject(error);
-        });
-    });
-  }
-  await window.placeholders[`${prefix}-loaded`];
-  return window.placeholders[prefix];
-}
-
-/**
- * Decorates a block.
- * @param {Element} block The block element
- */
-export function decorateBlock(block) {
-  const shortBlockName = block.classList[0];
-  if (shortBlockName) {
-    block.classList.add('block');
-    block.dataset.blockName = shortBlockName;
-    block.dataset.blockStatus = 'initialized';
-    const blockWrapper = block.parentElement;
-    blockWrapper.classList.add(`${shortBlockName}-wrapper`);
-    const section = block.closest('.section');
-    if (section) section.classList.add(`${shortBlockName}-container`);
-  }
+function decorateIcon(span, prefix = '') {
+  const iconName = Array.from(span.classList)
+    .find((c) => c.startsWith('icon-'))
+    .substring(5);
+  const img = document.createElement('img');
```

Comment:
@trieloff @kptdobe 

By using the `img` tag to load the SVG, you lose the ability to apply CSS to change its color and other properties. This is a common use case for small icons that for example should match the surrounding text. For example on https://blog.adobe.com/en/publish/2023/10/11/max-sneaks-2023, checkout the share icons. 

In some cases, using the `img` tag to load larger SVG illustrations makes sense as you don't want to increase the DOM size unnecessarily. 

So depending on the use case, ideally you should be able to decide whether you want to load the SVG with `<svg>` or with `<img>` WDYT ?

Reply:
Also this change should have been done in a separate PR IMO. It's hidden behind the franklin to aem rename currently.

Reply:
@icaraps We've had many issues and edge cases with the SVG sprite approach we had introduced earlier and the decision was taken to go back to a simple decoration that non-experts can leverage out-of-the-box for GA.
(check https://github.com/adobe/aem-lib/pull/15 for more details)

Since styling icons essentially requires that the SVG file be modified from the OOTB export that apps like Illustrator give you, it is assumed to be a developer use case that can be done directly at the project level for now. And we are planning to introduce a plugin system in the near future that would allow you to pull in the sprite logic back on-demand.

Reply:
@ramboz Ok thanks for the details. Do you have recommendations on how to best migrate projects using svg icons to images instead ? For example, the icon selector Sidekick plugin built for Nokia uses svg icons https://main--nokia--hlxsites.hlx.page/tools/sidekick/library.html?plugin=icons. Do you foresee any issues if they switch to use images instead ?

Reply:
@ramboz sounds good thanks !